### ISSUE TITLE
Search not restricted to sub-folder 

### ISSUE THREAD

**Issue by jnfrmarks** [owncloud/core#13687]
### Steps to reproduce

1. Install owncloud
2. Upload an odt or pdf file that has owncloud in it
3. Create a subfolder
4. Upload a bunch of odt or pdf files that have owncloud in them
5. Check the box next to the subfolder so that only that folder will be searched
6. Search from the root for owncloud

### Expected behaviour

Only files from the sub folder will be displayed in the search results 

### Actual behaviour

All files matching search results are displayed (from root and subfolder)

Ubuntu 14.04
PHP 5.5.9

### Server configuration

ownCloud Enterprise Edition 8.0 beta 2 (testing)

**Comment by PVince81** [comment]
@butonic

**Comment by DeepDiver1975** [comment]
@butonic please have a look if time permits - if not please ping me - THX

**Comment by MorrisJobke** [comment]
I guess the search is implemented as full search and not a "just in this subfolder" search. I guess this is actually a wanted behaviour and not a bug. 

Regarding UX: @jancborchardt Should the search just be in the subfolder or listing results from all places? I would vote for the subfolder approach as this is how it works on other platforms and therefore the users expect this behaviour.

**Comment by butonic** [comment]
limiting search to subfolders should wait on https://github.com/owncloud/core/issues/13155 as that will introduce the necessary changes on the server side.

**Comment by jnfrmarks** [comment]
Closing as not implemented yet.

**Comment by jancborchardt** [comment]
> Check the box next to the subfolder so that only that folder will be searched

That’s not really how file selection works with search. And I’m not sure people expect that this would be the case. It’s better to have a modeless search like now, prioritizing on the current folder but also taking into account subfolders / other places.

**Comment by jnfrmarks** [comment]
I have finally been able to get back to this. To close the loop on what was happening:

The search is actually searching the file names (I didn't realize this).  I chose to search for the word "owncloud" which was in the name of some of the files.

In 7.0.4, the results of the search is a drop down list of filenames containing the string.  In 8.0, the results of the search are displayed on a new page, which led me to believe that the results were of the files containing the search string and not filenames containing the search string.
